File: src/i18n/locales/en.ts
import type { Locale } from "../i18n.js";

export const en: Locale = {
  validation: {
    weight: {
      failedToParse: "Failed to parse weight format",
      invalidWeightNumberFormat:
        "Failed to recognize weight format. Expected a two- or three-digit number, optionally with a decimal part.",
      weightTooLow: "Weight cannot be less than 10 kg.",
      weightTooHigh: "Weight cannot be more than 200 kg.",
    },
    timestamp: {
      notString: "Timestamp is not a string.",
      notDate: "Failed to recognize date format.",
      failedToParseStart: "Failed to parse start date format",
      failedToParseEnd: "Failed to parse end date format",
    },
    auth: {
      failedToParse: "Failed to parse user data format",
      invalidFormat: "Username or password is not a string",
    },
  },
  response: {
    weight: {
      addSuccess: "Weight added successfully",
    },
    migration: {
      success: "Migration completed successfully",
    },
    user: {
      registerSuccess: "User registered successfully",
    },
  },
  error: {
    connection: {
      notSet: "Database connection not established",
    },
    user: {
      exists: "User with this name already exists",
      hashFailed: "Failed to hash the password",
      failedToAuthorize: "Authorization failed",
      unauthorized: "User is not authorized",
    },
    unknown: "An unknown error occurred",
  },
};

File: go/internal/i18n/locales/en.go
package locales

var En = Locale{
	Validation: ValidationLocale{
		Weight: WeightValidationLocale{
			FailedToParse:             "Failed to parse weight format",
			InvalidWeightNumberFormat: "Failed to recognize weight format. Expected a two- or three-digit number, optionally with a decimal part.",
			WeightTooLow:              "Weight cannot be less than 10 kg.",
			WeightTooHigh:             "Weight cannot be more than 200 kg.",
		},
		Timestamp: TimestampValidationLocale{
			NotString:          "Timestamp is not a string.",
			NotDate:            "Failed to recognize date format.",
			FailedToParseStart: "Failed to parse start date format",
			FailedToParseEnd:   "Failed to parse end date format",
		},
		Auth: AuthValidationLocale{
			FailedToParse: "Failed to parse user data format",
			InvalidFormat: "Username or password is not a string",
		},
	},
	Response: ResponseLocale{
		Weight:    "Weight added successfully",
		Migration: "Migration completed successfully",
		User:      "User registered successfully",
	},
	Error: ErrorLocale{
		Connection: ConnectionErrorLocale{
			NotSet: "Database connection not established",
		},
		User: UserErrorLocale{
			Exists:            "User with this name already exists",
			HashFailed:        "Failed to hash the password",
			FailedToAuthorize: "Authorization failed",
			Unauthorized:      "User is not authorized",
		},
		Unknown: "An unknown error occurred",
	},
}
